pkg/sdk/file_format_test.go

```diff
@@ -0,0 +1,316 @@
+package sdk
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFileFormatsCreate(t *testing.T) {
+	t.Run("minimal", func(t *testing.T) {
+		opts := &CreateFileFormatOptions{
+			name: NewSchemaObjectIdentifier("db1", "schema2", "format3"),
+			Type: FileFormatTypeCsv,
+		}
+		actual, err := structToSQL(opts)
+		require.NoError(t, err)
+		expected := `CREATE FILE FORMAT "db1"."schema2"."format3" TYPE = CSV`
+		assert.Equal(t, expected, actual)
+	})
+
+	t.Run("complete CSV", func(t *testing.T) {
+		opts := &CreateFileFormatOptions{
+			OrReplace:   Bool(true),
+			Temporary:   Bool(true),
+			name:        NewSchemaObjectIdentifier("db4", "schema5", "format6"),
+			IfNotExists: Bool(true),
+			Type:        FileFormatTypeCsv,
+
+			FileFormatTypeOptions: FileFormatTypeOptions{
+				CsvCompression:               &CsvCompressionBz2,
+				CsvRecordDelimiter:           String("-"),
+				CsvFieldDelimiter:            String(":"),
+				CsvFileExtension:             String("csv"),
+				CsvParseHeader:               Bool(true),
+				CsvSkipHeader:                Int(5),
+				CsvSkipBlankLines:            Bool(true),
+				CsvDateFormat:                String("YYYY-MM-DD"),
+				CsvTimeFormat:                String("HH:mm:SS"),
+				CsvTimestampFormat:           String("time"),
+				CsvBinaryFormat:              &BinaryFormatUtf8,
+				CsvEscape:                    String("\\"),
+				CsvEscapeUnenclosedField:     String("§"),
+				CsvTrimSpace:                 Bool(true),
+				CsvFieldOptionallyEnclosedBy: String("&"),
+				CsvNullIf: &[]NullString{
+					{"nul"},
+					{"nulll"},
```

Comment:
is 3 l's okay?

Reply:
yes the actual value doesn't matter, as long as it matches below. Users are meant to specify any special value they use for null.